lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -995,10 +996,10 @@ def payment_details(payment_method)
         case payment_method
         when String
           token_type_and_details(payment_method)
-        when NetworkTokenizationCreditCard
-          { payment_type: :network_token }
         else
-          { payment_type: :credit }
+          type = payment_method.respond_to?(:source) ? :network_token : :credit
```

Comment:
Can you add an assertion to an existing tests for Google Pay/Apple Pay that confirms that they are now going through the network tokenization path?

Reply:
Yes I can. The weird this is that they were going through the network tokenization path with the current way which makes me think that we are saving Google or Apple Pay as NetworkTokenizationCreditCard but as CreditCard